File: 001-010/007/007.py
#!/usr/bin/env python

# What is the 10 001st prime number?


def findprime():
    knownprimes = [2]
    i = 3
    while len(knownprimes) < 10001:
        for x in knownprimes:
            if i % x == 0:
                prime = False
                break
            else:
                prime = True
        if prime:
            knownprimes.append(i)
        i += 1
    return knownprimes[-1]


if __name__ == "__main__":
    print("The number is %d." % findprime())

File: 001-010/007/007.rb
#!/usr/local/bin/ruby -w

# What is the 10 001st prime number?

def findprime
    knownprimes = [2]
    i = 3
    prime = false

    while knownprimes.length < 10_001
        knownprimes.each do |x|
            if (i % x).zero?
                prime = false
                break
            else
                prime = true
            end
        end
        knownprimes.append(i) if prime
        i += 1
    end
    return knownprimes[-1]
end

puts "The number is #{findprime}."
